web-access-control/wac-allow/public-access-direct.feature

```diff
@@ -0,0 +1,91 @@
+Feature: The WAC-Allow header shows public access modes for a public agent when given direct access
+
+  Background: Create test resources giving a public agent different access modes
+    * table testModes
+      | test | modes | check |
+      | 'read' | ['read'] | 'only' |
+      | 'read/control' | ['read', 'control'] | 'only' |
+      | 'read/write' | ['read', 'write'] | '' |
+      | 'read/append' | ['read', 'append'] | 'only' |
+      | 'read/write/append' | ['read', 'write', 'append'] | 'only' |
+
+    * def setup =
+    """
+      function() {
+        const testContainer = createTestContainer();
+        const resources = {}
+        for (const row of testModes) {
+          const resource = testContainer.createChildResource('.ttl', karate.readAsString('../fixtures/example.ttl'), 'text/turtle');
+          if (resource.exists()) {
+            const access = resource.getAccessDatasetBuilder(webIds.alice)
+                  .setPublicAccess(resource.getUrl(), row.modes)
+                  .build();
+            resource.setAccessDataset(access);
+          }
+          resources[row.test] = resource;
+        }
+        return resources;
+      }
+    """
+    * def resources = callonce setup
+    * def resource = resources['read']
+
+  Scenario: There is an acl on the resource containing a public agent
+    Given url resource.aclUrl
+    And headers clients.alice.getAuthHeaders('GET', resource.aclUrl)
+    And header Accept = 'text/turtle'
+    When method GET
+    Then status 200
+    And match header Content-Type contains 'text/turtle'
+    # check array of triples contains one with the given predicate-object
+    # TODO: this would be better implemented as hasStatement(null, <http://www.w3.org/ns/auth/acl#agentClass> <http://xmlns.com/foaf/0.1/Agent>)
+    And match RDFUtils.turtleToTripleArray(response, resource.getUrl()) contains '#? _.includes("<http://www.w3.org/ns/auth/acl#agentClass> <http://xmlns.com/foaf/0.1/Agent>")'
+
+  Scenario: There is no acl on the parent that references a public agent
+    Given url resource.container.aclUrl
+    And headers clients.alice.getAuthHeaders('GET', resource.container.aclUrl)
+    And header Accept = 'text/turtle'
+    When method GET
+    Then assert responseStatus == 404
+
+  Scenario: Alice calls GET and the header shows full access for user
+    Given url resource.url
+    And headers clients.alice.getAuthHeaders('GET', resource.url)
+    When method GET
+    Then status 200
+    And match header WAC-Allow != null
+    * def result = parseWacAllowHeader(responseHeaders)
```

Comment:
What would happen if parsing fails? That should be a test failure too, I suppose?

Reply:
The header is not syntax checked but the parser simply returns as much as it can recover from the header so in the worst case, it returns `{ user: [], public: []}`. Subsequent tests of the content would then fail. Line 56 is important as it confirms the header is returned as line 57 would return the same for a bad header or a missing header.